lib/Data/Validate/Sanctions/Fetcher.pm

```diff
@@ -61,6 +61,13 @@ sub _ofac_xml {
         pattern  => '%m/%d/%Y',
         on_error => 'croak',
     );
+
+    # Pattern to look for: mm/dd/yyy
+    my $pattern = '((0[1-9]|1[012])[/]0[1-9]|[12][0-9]|3[01])[/]((19|20)\d\d)';
```

Comment:
better to move this to sub as its used here as well https://github.com/binary-com/perl-Data-Validate-Sanctions/pull/21/files#diff-f22d1e5548dc0825f328a4d72bbbcb96R90

Reply:
The regex for each case is different: one checks for `dd/mm/yyyy` and the other for `mm/dd/yyyy`

Reply:
I think this is overcomplicated. The pattern should be like `/(\d){1,2}/(\d){1,2}/(\d){4}/` 

for `dd/mm/yyyy` then you unpack the match to `($d, $m, $y) = ($1, $2, $3)` and for `mm/dd/yyyy` you just unpack by `($d, $m, $y) = ($2, $1,  $3)`, and only then it should try to instantiate `Date` to make sure it is correct.

The complex patterns do not provide needed defence agains bad dates, i.e. 30 of February, or 31 of November.